cdap-ui/app/cdap/components/EntityCard/index.js

```diff
@@ -110,6 +115,56 @@ export default class EntityCard extends Component {
         successMessage={this.state.successMessage}
       />
     );
+
+    let card = (
+      <Card
+        header={header}
+        id={classnames(
+          this.props.entity.isHydrator
+            ? `entity-cards-datapipeline`
+            : `entity-cards-${this.props.entity.type}`
+        )}
+        onClick={this.onClick.bind(this)}
+      >
+        <div className="entity-information clearfix">
+          <div
+            className={classnames('entity-id-container', {
+              'with-version': this.props.entity.version,
+            })}
+          >
+            <h4
+              className={classnames({ 'with-version': this.props.entity.version })}
+              title={this.props.entity.id}
+            >
+              {this.props.entity.id}
+            </h4>
+            <small>{this.props.entity.version ? this.props.entity.version : ''}</small>
+          </div>
+        </div>
+
+        {this.renderEntityStatus()}
+
+        <div className="fast-actions-container text-center">
+          <FastActions
+            entity={this.props.entity}
+            onUpdate={this.onFastActionUpdate.bind(this)}
+            onSuccess={this.props.onFastActionSuccess}
+            className="btn-group"
+          />
+        </div>
+      </Card>
+    );
+
+    if (this.props.entity.isHydrator) {
+      const namespace = getCurrentNamespace();
+      const link = `/pipelines/ns/${namespace}/view/${this.props.entity.id}`;
```

Comment:
I would suggest getting this link using url-generator that we have to go to `hydrator.detail`